### PR TITLE
Prevent timeouts for a pgwire conn's txns for as long as it stays open

### PR DESCRIPTION
**Pull request by danhhz**

### CODE REVIEW

File: kv/txn_coord_sender.go [hunk 1]
```diff
@@ -550,6 +551,17 @@ func (tc *TxnCoordSender) unregisterTxnLocked(txnID uuid.UUID) (duration, restar
 	return
 }
 
+func (tc *TxnCoordSender) TxnPing(txnID uuid.UUID) {
+	txnMeta := tc.txns[txnID]
```
Comment (tbg) [inline kv/txn_coord_sender.go:555]: this already needs the lock.

File: kv/txn_coord_sender.go [hunk 2]
```diff
@@ -550,6 +551,17 @@ func (tc *TxnCoordSender) unregisterTxnLocked(txnID uuid.UUID) (duration, restar
...
+	if txnMeta != nil {
+		tc.Lock()
+		nowNanos := tc.clock.PhysicalNow()
+		txnMeta.setLastUpdate(nowNanos)
+		tc.Unlock()
+		log.Infof("TxnPing set last update to %d\n", nowNanos)
```
Comment (tbg) [inline kv/txn_coord_sender.go:561]: this is likely going to be too noisy. `if log.V(1) { }` perhaps.

File: sql/pgwire/v3.go [hunk 3]
```diff
@@ -157,6 +160,25 @@ func (c *v3Conn) parseOptions(data []byte) error {
 }
 
 func (c *v3Conn) serve(authenticationHook func(string, bool) error) error {
+	var tickChan <-chan time.Time
+	{
+		ticker := time.NewTicker(storage.DefaultHeartbeatInterval / 2)
+		tickChan = ticker.C
+		defer ticker.Stop()
+	}
+	go func() {
```
Comment (tbg) [inline sql/pgwire/v3.go:169]: Hmm, this extra goroutine makes me think that maybe we should be completely event-driven - as long as `serve` is active, the txn does not time out. When the conn goes down, we call "Unping" (so basically "hey this thing is abandoned right now").

**Comment by tbg** [comment]
Yeah, this is the right direction. I'm about to run out for an interview, so I my feedback is somewhat half-baked.

File: sql/pgwire/v3.go (hunk 3 above)
Comment (tamird) [inline sql/pgwire/v3.go:169]: +1 use a defer for this

**Pull request closed without merging**